cmd/cored/main.go

```diff
@@ -233,7 +234,21 @@ func launchConfiguredCore(ctx context.Context, db *sql.DB, conf *config.Config,
 		if err != nil {
 			chainlog.Fatal(ctx, chainlog.KeyError, err)
 		}
-		s := blocksigner.New(blockPub, mockHSM, db, c) // TODO(jackson): support real HSM
+
+		var hsm blocksigner.Signer = mockHSM
+		if conf.BlockHSMURL != "" {
+			//TODO(ameets): these members may not all be needed - reevaluate
```

Comment:
All of them at least add information to the user agent or header which can be helpful in debugging. I think it's fine to leave all of them and clear this todo.